analysis/study_definition.py

```diff
@@ -1,13 +1,884 @@
+      find_last_match_in_period=True,
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.01,
+          "rate": "exponential_increase",
+      },
+  )
+def drug_6m(dx_codelist):
+  return patients.with_these_medications(
+      dx_codelist,
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 6 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.01,
+      },
+  )
+def drug_12m(dx_codelist):
+  return patients.with_these_medications(
+      dx_codelist,
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 12 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.01,
+      },
+  )
+def high_cost_drugs_3m(dx_codelist):
+  return patients.with_high_cost_drugs(
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 3 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.01,
+      },
+  ) 
+def high_cost_drugs_6m(dx_codelist):
+  return patients.with_high_cost_drugs(
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 6 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.01,
+      },
+  ) 
+def covid_therapeutics(dx_codelist):
+  return patients.with_covid_therapeutics(
+      dx_codelist,
+      with_these_indications = "non_hospitalised",
+      between = ["covid_test_positive_date", "covid_test_positive_date + 5 days"],
+      find_first_match_in_period=True,
+      returning="date",
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.9,
+          "date": {"earliest": "2021-12-16"},
+      },
+  )
+def adverse_outcome_icd(dx_codelist):
+  return patients.admitted_to_hospital(
+      with_these_diagnoses = dx_codelist,
+      returning="date_admitted",
+      between = ["covid_test_positive_date", "covid_test_positive_date + 28 days"],
+      find_first_match_in_period=True,
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.2,
+          "date": {"earliest": "2021-12-16"},
+      },
+  )
+def adverse_outcome_snomed(dx_codelist):
+  return patients.with_these_clinical_events(
+      dx_codelist,
+      returning="date",
+      between = ["covid_test_positive_date", "covid_test_positive_date + 28 days"],
+      find_first_match_in_period=True,
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.2,
+          "date": {"earliest": "2021-12-16"},
+      },
+  )
+def imae_snomed(dx_codelist):  ## to idenitfy NEW conditions, will include all first diagnosis but then on stata remove those diagnosed before covid test positive date
```

Comment:
Why not exclude here i.e., use `between` rather than `on_or_before`? Or are you also interested in diagnoses prior to testing positive?

Reply:
Yes, we are interested in defining them as new. So i assumed, to ensure if it a first ever code, i wanted to include all first diagnosis but then on Stata remove those diagnosed before Covid test positive date. 
Is there a better way to do this?

Reply:
Oh I see. It is possible to do this within the study def - worth asking this question in the opensafely users channel eve if you don't end up using the answer and do it post extraction.